src/stamp/encoding/encoder/__init__.py

```diff
@@ -0,0 +1,193 @@
+        self,
+        output_dir: Path,
+        feat_dir: Path,
+        device: DeviceLikeType,
+        generate_hash: bool,
+        **kwargs,
+    ) -> None:
+        """General method to encode slides."""
+        output_file = self._generate_output_path(
+            output_dir=output_dir, generate_hash=generate_hash
+        )
+
+        slide_dict = {}
+        self.model.to(device).eval()
+
+        if self.precision == torch.float16:
+            self.model.half()
+
+        if os.path.exists(output_file):
+            tqdm.write(f"Output file {output_file} already exists, skipping")
+            return
+
+        for tile_feats_filename in tqdm(os.listdir(feat_dir), desc="Processing slides"):
+            h5_path = os.path.join(feat_dir, tile_feats_filename)
+            slide_name: str = Path(tile_feats_filename).stem
+
+            try:
+                feats, coords = self._validate_and_read_features(h5_path)
+            except ValueError as e:
+                tqdm.write(s=str(e))
+                continue
+
+            slide_embedding = self._generate_slide_embedding(
+                feats, device, coords=coords
+            )
+            slide_dict[slide_name] = {"feats": slide_embedding}
+
+        self._save_features(output_file, entry_dict=slide_dict)
+
+    def encode_patients(
+        self,
+        output_dir: Path,
+        feat_dir: Path,
+        slide_table_path: Path,
+        patient_label: PandasLabel,
+        filename_label: PandasLabel,
+        device: DeviceLikeType,
+        generate_hash: bool,
+        **kwargs,
+    ) -> None:
+        """General method to encode patients with features only.
+        Override this method if coords are needed for encoding"""
+        output_file = self._generate_output_path(
+            output_dir=output_dir, generate_hash=generate_hash
+        )
+
+        patient_dict = {}
+        self.model.to(device).eval()
+
+        if self.precision == torch.float16:
+            self.model.half()
+
+        if os.path.exists(output_file):
+            tqdm.write(f"Output file {output_file} already exists, skipping")
+            return
+
+        slide_table = self._read_slide_table(slide_table_path)
+        patient_groups = slide_table.groupby(patient_label)
+
+        for patient_id, group in tqdm(patient_groups, leave=False):
+            feats_list = []
+
+            for _, row in group.iterrows():
+                slide_filename = row[filename_label]
+                h5_path = os.path.join(feat_dir, slide_filename)
+                feats, _ = self._validate_and_read_features(h5_path)
+                feats_list.append(feats)
+
+            if not feats_list:
+                tqdm.write(f"No features found for patient {patient_id}, skipping.")
+                continue
+
+            patient_embedding = self._generate_patient_embedding(
+                feats_list, device, **kwargs
+            )
+            patient_dict[patient_id] = {"feats": patient_embedding}
```

Comment:
Are we storing other information apart from feats in here? If not, why don't we just do:
```suggestion
            patient_dict[patient_id] = patient_embedding
```

Then the type of `patient_dict` would be `dict[str, np.ndarray]`